### ISSUE TITLE
Help Needed for metric type summary

### ISSUE THREAD

**Issue by miteshvp** [rycus86/prometheus_flask_exporter#29]
I have two questions
1. Is there a way to store average latency time based on the histogram `flask_http_request_duration_seconds_sum` and `flask_http_request_duration_seconds_count`  as another metric?

2. What is the best way to pass r.status_code if you do not use `@app.route` to instantiate? I am using BluePrint and Flask_restful to register and instantiate separately as in

Following your example on creating a summary
```
@app.route('/status/<int:status>')
@metrics.summary('requests_by_status', 'Request latencies by status', labels={'status': lambda r: r.status_code})
        def echo_status(status):
            return 'Status: %s' % status, status
```

```
from flask import Blueprint
from flask_restful import  Resource

api_v1 = Blueprint('api_v1', __name__, url_prefix='/api/v1')
rest_api_v1 = Api(api_v1, errors=errors)


def add_resource(resource, route, endpoint=None, defaults=None):
    """Add a resource for both trailing slash and no trailing slash to prevent redirects."""
    slashless = route.rstrip('/')
    endpoint = endpoint or resource.__name__.lower()
    defaults = defaults or {}

    # resources without slashes
    rest_api_v1.add_resource(resource,
                             slashless,
                             endpoint=endpoint + '__slashless',
                             defaults=defaults)

class Test(Resource):
    status = 200

    @staticmethod
    @metrics.summary('dummy_test', 'Test Request latencies by status',
                     labels={'dummy_test': lambda r: ???})
    def get():
        try:
            time.sleep(random.randint(0, 5))
            if random.choice([True, False]):
                return 'OK', 200
            else:
                return 'Not OK', 400
        except Exception as e:
            logger.error('%r' % e)

add_resource(Test, '/test', endpoint='test')
```

**Comment by rycus86** [comment]
Good questions!

On the average, I don't think there's anything built-in in the underlying library, though it should be just `*_sum / *_count` on the other end, or something similar? If you would want to compute it in Flask, would it be a total average since the app has started, or a rolling average for the last X minutes, or something else? I've seen this calculated on the aggregation system more, like in Grafana, which averages for you based on the samples received in a given time period.

For the second question, if you can't access the response there as expected, then I'll need to investigate. I haven't used Blueprints much. I can try your example over the weekend.

Thanks!

**Comment by miteshvp** [comment]
Thanks for your quick reply. I was thinking of calculating average since lifetime (because _sum and _count are probably since the start of a process).
I will wait to hear from you for the second point.
Thanks.

**Comment by rycus86** [comment]
Tried to have a look today, it looks like I'd need to add some `Flask-RESTful` specific heuristics in the code to detect if we're currently in a restful handler function or not. Should be doable, but it is a bit complicated, so I'll have to have another look to see what's the best way of supporting that.

**Comment by rycus86** [comment]
I think I managed to get it working for your use case. It's running through Travis now, and should be available in `0.8.2` very soon. Thanks again for the report!

**Comment by miteshvp** [comment]
Thank you. I will give it a try on `0.8.2`
